### PR TITLE
Simple full text search on pages

### PR DESCRIPTION
**Pull request by lecunm**

### CODE REVIEW

File: app/services/SearchService.scala [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+  def getHierarchyPath(pageJoinProject: PageJoinProject): String = {
+    val projectHierarchies = hierarchyRepository.findAllByProjectId(pageJoinProject.project.id)
+    if (projectHierarchies.nonEmpty) {
+      val hierarchyId = projectHierarchies.head.id
+      val splitedId = hierarchyId.split('.')
+      var hierarchyPath = ""
+      var currentHierarchyId = "."
+      for (id <- splitedId) {
+        if (id != "") {
```
Comment (notflorian) [inline app/services/SearchService.scala:55]:
```suggestion
        if (id.nonEmpty) {
```

File: app/services/SearchService.scala [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+case class PageIndexDocument(hierarchy: String, path: String, branch: String, label: String, description: String, pageContent: String)
+
+@Singleton
+class PageIndex {
+
+  val luceneSearchIndex = new DirectLucene(uniqueFields = List(), defaultFullTextSearchable = true)
+
+  val hierarchy = luceneSearchIndex.create.field[String]("hierarchy")
+  val path = luceneSearchIndex.create.field[String]("path", fullTextSearchable = false)
```
[the remark this answers is not in the document]
Comment (lecunm) [reply app/services/SearchService.scala:18]: yes we want the full text search on hierarchy and branch name

File: app/services/SearchService.scala [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+}
+
+class SearchService @Inject()(pageIndex: PageIndex, hierarchyRepository: HierarchyRepository) {
+
+  def searchForPage(keywords: String): Seq[PageIndexDocument] = {
+    pageIndex.query(keywords)
+  }
+
+  def getHierarchyPath(pageJoinProject: PageJoinProject): String = {
```
[the remark this answers is not in the document]
Comment (lecunm) [reply app/services/SearchService.scala:47]: it wasn't at the right place

File: app/services/SearchService.scala [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+import models.PageJoinProject
+import repositories.HierarchyRepository
+
+
+case class PageIndexDocument(hierarchy: String, path: String, branch: String, label: String, description: String, pageContent: String)
+
+@Singleton
+class PageIndex {
+
+  val luceneSearchIndex = new DirectLucene(uniqueFields = List(), defaultFullTextSearchable = true)
```
[the remark this answers is not in the document]
Comment (lecunm) [reply app/services/SearchService.scala:15]: unique fields is a unique identifier and we decided that it will be the path as we were doing in the cache

**Pull request merged**